src/query-builder/QueryBuilder.ts

```diff
@@ -993,51 +993,68 @@ export abstract class QueryBuilder<Entity> {
 
             if (metadata.hasRelationWithPropertyPath(path)) {
                 const relation = metadata.findRelationWithPropertyPath(path)!;
+                const isEntityTarget = relation.inverseEntityMetadata.target === entity[key].constructor;
 
-                // There's also cases where we don't want to return back all of the properties.
-                // These handles the situation where someone passes the model & we don't need to make
-                // a HUGE `where` to uniquely look up the entity.
+                const primaryColumns = relation.inverseEntityMetadata.primaryColumns;
+                const seenPrimaryColumns = new Set<string>();
+                for (const primaryColumn of primaryColumns) {
+                    const entityValueMap = primaryColumn.getEntityValueMap(entity[key]);
+                    for (const [columnName, value] of Object.entries(entityValueMap || {})) {
+                        if (value) {
```

Comment:
Is `null` a valid PK value?  I know `0` is.  So is an empty string.  Is that something we should be handling here?  Only checking for `undefined` instead?